### PR TITLE
make internal variables in static_rand unsigned.

### PR DESCRIPTION
**Pull request by CasparKielwein**
several of the bitwise manipulations on signed integers are technically undefined behaviour.
I made the internal variables unsigned and the results are then converted to signed ints as before.
This should not actually change any observable results.

Did some cleanup as well, while I was at it.

Fixes the first part of  #1210

### CODE REVIEW

File: code/math/staticrand.cpp [hunk 1]
```diff
@@ -7,26 +7,21 @@
...
 
-
-
 #include "math/staticrand.h"
 #include "math/vecmat.h"
 
-
-int Semirand_inited = 0;
+bool Semirand_inited = false;
 int Semirand[SEMIRAND_MAX];
```
Comment (MageKing17) [inline code/math/staticrand.cpp]: If this is only used internally, in code sections that now constantly convert it to unsigned, why not just make it an array of unsigned ints?

Comment (CasparKielwein) [reply code/math/staticrand.cpp]: true,
in my first try, I converted the whole interface to unsigned.
When I reverted that, I missed this bit.

File: code/math/staticrand.cpp [hunk 2]
```diff
@@ -37,19 +32,20 @@ void init_semirand()
...
-	c = (num >> (2 * SEMIRAND_MAX_LOG)) & (SEMIRAND_MAX - 1);
+	const unsigned int num_unsigned = num;
+	auto a = num_unsigned & (SEMIRAND_MAX - 1);
+	auto b = (num_unsigned >> SEMIRAND_MAX_LOG) & (SEMIRAND_MAX - 1);
+	auto c = (num_unsigned >> (2 * SEMIRAND_MAX_LOG)) & (SEMIRAND_MAX - 1);
 
-	return Semirand[a] ^ Semirand[b] ^ Semirand[c];
+	return Semirand[a]
+			^ Semirand[b]
+			^ Semirand[c];
```
Comment (MageKing17) [inline code/math/staticrand.cpp]: These lines are now spread out for no reason.

Comment (CasparKielwein) [reply code/math/staticrand.cpp]: narf, I'm to used to having an auto-formatter on save.

**Pull request merged**